de-lib/src/main/java/org/iplantc/de/analysis/client/presenter/AnalysesPresenterImpl.java

```diff
@@ -448,7 +475,83 @@ public void onSuccess(AnalysisStepsInfo result) {
+        asr.setUser(deProperties.getSupportUser());
+        asr.setAnalysisPermissions(Arrays.asList(ap));
+        AnalysisSharingRequestList listRequest = shareFactory.AnalysisSharingRequestList().as();
+        listRequest.setAnalysisSharingRequestList(Arrays.asList(asr));
+        analysisService.shareAnalyses(listRequest, new AnalysisCallback<String>() {
+            @Override
+            public void onFailure(Integer statusCode, Throwable exception) {
+                ErrorHandler.post(exception);
+            }
+
+            @Override
+            public void onSuccess(String result) {
+                emailSupport(parent);
+            }
+        });
+    }
+
+    @Override
+    public void onUserSupportRequested(AnalysisUserSupportRequestedEvent event) {
+        final Analysis value = event.getValue();
+        final AnalysisUserSupportDialog ausd = new AnalysisUserSupportDialog(value);
+        ausd.setHeadingHtml(value.getName());
+        ausd.setSize("800px", "500px");
+        ausd.addSubmitSelectHandler(new SelectEvent.SelectHandler() {
+            @Override
+            public void onSelect(SelectEvent event) {
+                Splittable parent = StringQuoter.createSplittable();
+                final Splittable sp = StringQuoter.createSplittable();
+                StringQuoter.create(value.getName()).assign(sp, "Name");
+                StringQuoter.create(value.getAppName()).assign(sp, "App");
+                StringQuoter.create(value.getResultFolderId()).assign(sp, "Output Folder");
+                StringQuoter.create(new Date(value.getStartDate()).toString()).assign(sp, "Start Date");
+                StringQuoter.create(new Date(value.getEndDate()).toString()).assign(sp, "End Date");
+                StringQuoter.create(ausd.getComment()).assign(sp, "Comment");
+                StringQuoter.create(value.getStatus()).assign(sp, "Status");
+                StringQuoter.create(UserInfo.getInstance().getEmail()).assign(sp, "Email");
+
+                StringQuoter.create(UserInfo.getInstance().getFirstName() + " " + UserInfo.getInstance()
+                                                                                          .getLastName())
+                            .assign(sp, "User");
+                sp.assign(parent, "fields");
+
+                StringQuoter.create(UserInfo.getInstance().getFullUsername()).assign(parent, "from");
+                StringQuoter.create(
+                        UserInfo.getInstance().getUsername() + " requesting help with Analysis")
+                            .assign(parent, "Subject");
+                GWT.log("splittable -->" + parent.getPayload());
+                ausd.hide();
+                shareWithSupport(value, parent);
+            }
+        });
+
+        ausd.show();
+
+    }
+
+    protected void emailSupport(Splittable parent) {
+        supportService.submitSupportRequest(parent, new AsyncCallback<Void>() {
```

Comment:
(here's the other spot it'd need to be changed to use an `AsyncProviderWrapper`, I think)